### PR TITLE
chore: don't check searcher shutdown to trigger work hp importance work

### PR DESCRIPTION
**Pull request by stoksc**
## Description
Exactly what it says on the tin.

## Commentary (optional)
I really don't think this should make a difference, but I don't have a lot of time to look and we can expend no effort to see if it works.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: master/internal/experiment.go [hunk 1]
```diff
@@ -297,10 +299,6 @@ func (e *experiment) Receive(ctx *actor.Context) error {
 			experiment:     e.Experiment,
 		})
 
-		if e.searcher.Shutdown {
```
Comment (stoksc) [inline master/internal/experiment.go:300]: didn't move this up for any particular reason.

File: master/internal/experiment.go [hunk 2]
```diff
@@ -284,6 +284,8 @@ func (e *experiment) Receive(ctx *actor.Context) error {
 		}
 		telemetry.ReportExperimentStateChanged(ctx.Self().System(), e.db, *e.Experiment)
 
+		ctx.Tell(e.hpImportance, hpimportance.ExperimentCompleted{ID: e.ID})
+
```
Comment (mackrorysd) [inline master/internal/experiment.go:288]: Didn't you hit a panic before, that wouldn't be prevent by this re-ordering?

Comment (mackrorysd) [reply master/internal/experiment.go:288]: I just tested changing the e.Searcher.shutdown conditional to `if e.State == model.CompletedState {` and it seems to work well, but I don't know exactly what you do to hit all the panics you did.

Comment (stoksc) [reply master/internal/experiment.go:288]: I think it should be handled by https://github.com/determined-ai/determined/pull/2230/files#diff-f8f79514e8276c806fcfa81745d24ada457e637648894ebf1d006ac5d4a29e46R188 now.

>  just tested changing the e.Searcher.shutdown conditional to if e.State == model.CompletedState { and it seems to work well, but I don't know exactly what you do to hit all the panics you did.

That is super strange..

**Review by mackrorysd — APPROVED** [review]

**Review by mackrorysd — APPROVED** [review]

**Pull request merged**